### PR TITLE
[release-1.3] support telemetry addons (#286)

### PR DESCRIPTION
**Pull request by ostromart**
* Support ThirdParty addOns in Operator

* Fix lint and test error

* Fix Test

* Fix lint error

### CODE REVIEW

**Review by rlenglet — CHANGES_REQUESTED** [review]
Please link to the `master` PRs that are backported here.

**Comment by irisdingbj** [comment]
@rlenglet THis is the PR for master: https://github.com/istio/operator/pull/286

File: cmd/mesh/testdata/manifest-generate/output/pilot_override_kubernetes.yaml [hunk 1]
```diff
@@ -190,6 +190,32 @@ spec:
     istio: pilot
 
 ---
+apiVersion: policy/v1beta1
```
Comment (rlenglet) [inline cmd/mesh/testdata/manifest-generate/output/pilot_override_kubernetes.yaml]: This file wasn't modified in https://github.com/istio/operator/pull/286. What PR modified it in `master`?

File: cmd/mesh/testdata/manifest-generate/output/pilot_override_values.yaml [hunk 2]
```diff
@@ -643,6 +643,478 @@ metadata:
     release: istio
 ---
 
+# Resources for Prometheus component
```
Comment (rlenglet) [inline cmd/mesh/testdata/manifest-generate/output/pilot_override_values.yaml]: This file wasn't modified in https://github.com/istio/operator/pull/286. What PR modified it in `master`?

File: cmd/mesh/testdata/manifest-generate/output/telemetry_k8s_settings.yaml [hunk 3]
```diff
@@ -1,3 +1,475 @@
+# Resources for Prometheus component
```
Comment (rlenglet) [inline cmd/mesh/testdata/manifest-generate/output/telemetry_k8s_settings.yaml]: This file wasn't modified in https://github.com/istio/operator/pull/286. What PR modified it in `master`?

File: cmd/mesh/testdata/manifest-generate/output/telemetry_override_kubernetes.yaml [hunk 4]
```diff
@@ -1,3 +1,475 @@
+# Resources for Prometheus component
```
Comment (rlenglet) [inline cmd/mesh/testdata/manifest-generate/output/telemetry_override_kubernetes.yaml]: This file wasn't modified in https://github.com/istio/operator/pull/286. What PR modified it in `master`?

File: cmd/mesh/testdata/manifest-generate/output/telemetry_override_values.yaml [hunk 5]
```diff
@@ -1,3 +1,475 @@
+# Resources for Prometheus component
```
Comment (rlenglet) [inline cmd/mesh/testdata/manifest-generate/output/telemetry_override_values.yaml]: This file wasn't modified in https://github.com/istio/operator/pull/286. What PR modified it in `master`?

**Review by rlenglet — CHANGES_REQUESTED** [review]

File: cmd/mesh/testdata/manifest-generate/output/pilot_override_kubernetes.yaml (hunk 1 above)
Comment (ostromart) [reply cmd/mesh/testdata/manifest-generate/output/pilot_override_kubernetes.yaml]: The charts and values in istio/installer are not identical between master and release-1.3, which leads to differences in the golden outputs. This is also why often patches do not apply cleanly and must be manually merged.

Comment (rlenglet) [reply cmd/mesh/testdata/manifest-generate/output/pilot_override_kubernetes.yaml]: This was not my question. I need to be able to trace backported changes back to their original `master` PR. So in which `master` PR was this change made?

**Review by rlenglet — APPROVED** [review]
I looked up the `master` PRs myself. LGTM.
Next time, please list all the original `master` PRs in the backport PR description.

**Comment by ostromart** [comment]
Thanks - sorry I was actually looking into this and found another problem that I got stuck on. There's a test flake going on which is confusing things.

**Pull request merged**